targets/TARGET_STM/sleep.c

```diff
@@ -236,25 +223,43 @@ __WEAK void hal_deepsleep(void)
      * us_ticker timestamp until the us_ticker context is restored. */
     mbed_sdk_inited = 0;
 
-    /* We've seen unstable PLL CLK configuration when DEEP SLEEP exits just few µs after being entered
-    *  So we need to force clock init out of Deep Sleep.
-    *  This init has been split into 2 separate functions so that the involved structures are not allocated on the stack in parallel.
-    *  This will reduce the maximum stack usage in case on non-optimized / debug compilers settings
-    */
-    ForceOscOutofDeepSleep();
-    ForcePeriphOutofDeepSleep();
-
     /* After wake-up from STOP reconfigure the PLL */
 #if defined(DUAL_CORE)
-    while (LL_HSEM_1StepLock(HSEM, CFG_HW_RCC_SEMID)) {
+    /* CFG_HW_STOP_MODE_SEMID is used to protect read access to STOP flag, and this avoid both core to configure clocks if both exit from stop at the same time */
+    while (LL_HSEM_1StepLock(HSEM, CFG_HW_STOP_MODE_SEMID)) {
     }
 
-    if ((LL_RCC_GetSysClkSource() ==  LL_RCC_SYS_CLKSOURCE_STATUS_HSI)) {
-        LL_PWR_ClearFlag_CPU();
+    /* Clocks need to be reconfigured only if system has been in stop mode */
+    if (LL_PWR_CPU_IsActiveFlag_STOP() && LL_PWR_CPU2_IsActiveFlag_STOP()) {
+        /* We've seen unstable PLL CLK configuration when DEEP SLEEP exits just few µs after being entered
+        *  So we need to force clock init out of Deep Sleep.
+        *  This init has been split into 2 separate functions so that the involved structures are not allocated on the stack in parallel.
+        *  This will reduce the maximum stack usage in case on non-optimized / debug compilers settings
+        */
+        while (LL_HSEM_1StepLock(HSEM, CFG_HW_RCC_SEMID)) {
+        }
+        ForceOscOutofDeepSleep();
+        ForcePeriphOutofDeepSleep();
         SetSysClock();
+        LL_HSEM_ReleaseLock(HSEM, CFG_HW_RCC_SEMID, HSEM_CR_COREID_CURRENT);
     }
-    LL_HSEM_ReleaseLock(HSEM, CFG_HW_RCC_SEMID, HSEM_CR_COREID_CURRENT);
+#if defined(CORE_CM7)
```

Comment:
Isn't there a define for CPU1 / CPU2  ? 
just in case another DUAL_CORE than H7 may have another split ...

Reply:
No, there is no such defined.
I entered an internal change request: 
ST_INTERNAL_REF  76709